estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,67 @@
+from odoo import models, fields, api
+from dateutil.relativedelta import relativedelta
+from odoo.exceptions import UserError
+
+
+class EstatePropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Property Offer'
+    _order = 'price desc'
+
+    _check_expected_price = models.Constraint(
+        'CHECK(price >= 0)',
+        'An offer price must be strictly positive',
+    )
+
+    price = fields.Float(required=True)
+    status = fields.Selection(
+        selection=[('accepted', 'Accepted'), ('refused', 'Refused'), ('pending', 'Pending')],
+        default='pending',
+    )
+
+    partner_id = fields.Many2one('res.partner', string="Partner", required=True)
+    property_id = fields.Many2one('estate.property', string="Property", required=True)
+    property_type_id = fields.Many2one(related='property_id.type_id', string="Property Type", store=True)
+    property_state = fields.Selection(related='property_id.state', string="Property State")
+    property_expected_price = fields.Float(related='property_id.expected_price', string="Expected Price")
+
+    create_date = fields.Datetime(readonly=True, default=fields.Datetime.now)
+    validity = fields.Integer(default=7)
+    date_deadline = fields.Date(
+        compute='_compute_deadline',
+        inverse='_inverse_deadline',
+        string="Deadline")
+
+    @api.model
```

Comment:
```suggestion
    @api.model_create_multi
```
This ensures that you always get a List[Dict[str, Any]] in vals_list